### PR TITLE
[CI] Configured travis builds #15

### LINKED ISSUE #15: [ci] Setup travis build with code coverage
Follow best practices used in openwisp regarding CI builds.

A recent similar work we did was in openwisp-firmware-upgrade: https://github.com/openwisp/openwisp-firmware-upgrader/blob/master/.travis.yml

### PR DESCRIPTION
**Pull request by TheOneAboveAllTitan**
Closes #15

`pip` was erroring out due to incompatible requirements for `cryptography` module, so had to mention it in `requirements.txt`. [See for more info](https://travis-ci.com/github/TheOneAboveAllTitan/openwisp-monitoring/builds/158418833#L547)

Kindly suggest any other way to mitigate this problem.
@nemesisdesign please turn on Travis and Coveralls for this repository.

### CODE REVIEW

File: requirements-test.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+coveralls
```
Comment (nemesisdesign) [inline requirements-test.txt:1]: not needed, openwisp-utils brings this dependency in: https://github.com/openwisp/openwisp-utils/blob/master/requirements-test.txt#L1

File: .travis.yml [hunk 2]
```diff
@@ -1,34 +1,48 @@
...
+  - pip install --no-cache-dir -U -r requirements.txt -r requirements-test.txt
 
 install:
   - python setup.py -q develop
 
 script:
+  - |
+    openwisp-utils-qa-checks \
+          --migration-path ./openwisp_monitoring/monitoring \
+          --migration-module monitoring
```
Comment (nemesisdesign) [inline .travis.yml:44]: --migration-module monitoring monitoring device check

File: .travis.yml [hunk 3]
```diff
@@ -1,34 +1,48 @@
...
     - master
 
 before_install:
+  - docker run -d --name influxdb -p 8086:8086 influxdb
+  - docker exec -it influxdb influx -execute 'CREATE DATABASE openwisp2'
   - pip install -U pip wheel setuptools
-  - pip install --no-cache-dir -U -r requirements-test.txt
-  - ./runflake8
-  - ./runisort
+  - pip install --no-cache-dir -U -r requirements.txt -r requirements-test.txt
```
Comment (nemesisdesign) [inline .travis.yml:35]: install only `requirements-test.txt`, don't worry if it fails, I will help you debug it.

File: .travis.yml (hunk 2 above)
Comment (TheOneAboveAllTitan) [reply .travis.yml]: I have update accordingly, took reference from [openwisp-controller](https://github.com/openwisp/openwisp-controller/blob/732c4e4667e00b6b654a785f6092a781676efed3/.travis.yml#L47)

File: requirements-test.txt (hunk 1 above)
Comment (TheOneAboveAllTitan) [reply requirements-test.txt]: But that's in `requirements-test.txt` :sweat_smile:

**Comment by nemesisdesign** [comment]
@nepython coveralls is pulled from `openwisp-utils[qa]` in `requirements.txt`: https://github.com/openwisp/openwisp-monitoring/blob/master/requirements-test.txt

File: .travis.yml [hunk 4]
```diff
@@ -1,34 +1,53 @@
...
 branches:
   only:
     - master
 
 before_install:
-  - pip install -U pip wheel setuptools
+  - docker run -d --name influxdb -p 8086:8086 influxdb
+  - docker exec -it influxdb influx -execute 'CREATE DATABASE openwisp2'
+  - pip install -U pip wheel setuptools pyasn1
   - pip install --no-cache-dir -U -r requirements-test.txt
```
Comment (nemesisdesign) [inline .travis.yml:36]: let's remove `--no-cache-dir` for consistency with other modules. If you find this in other modules let me know so we can remove it. I added it at some point in the past in which I was going crazy with the travis cache. Then they added a way to clear the cache from the web app.

File: .travis.yml [hunk 5]
```diff
@@ -1,34 +1,53 @@
...
-  - ./runflake8
-  - ./runisort
 
 install:
   - python setup.py -q develop
```
Comment (nemesisdesign) [inline .travis.yml:39]: let's use `pip install -e .`

File: .travis.yml [hunk 6]
```diff
@@ -1,34 +1,53 @@
...
 script:
+  - |
+    openwisp-utils-qa-checks \
+          --migration-path \
+          "./openwisp_monitoring/check/migrations \
+          ./openwisp_monitoring/device/migrations \
+          ./openwisp_monitoring/monitoring/migrations \
+          ./openwisp_monitoring/notifications/migrations" \
+          --migration-module "check device_monitoring monitoring notifications"
   - coverage run --source=openwisp_monitoring runtests.py
```
Comment (nemesisdesign) [inline .travis.yml:50]: I understood why this is failing, can you add `pip install https://github.com/openwisp/openwisp-utils/tarball/master` temporarily in `before_install` please?

File: requirements.txt [hunk 7]
```diff
@@ -5,3 +5,4 @@ django-apptemplates>=1.4,<2.0
 django-celery-email>=3.0.0,<3.1
 djangorestframework>=3.11,<3.12
 swapper
+cryptography<2.9.0,>=2.4.0
```
Comment (nemesisdesign) [inline requirements.txt:8]: not sure why this is needed, it should be already specified somewhere upstream. Can you try commenting it out and let the build run without it please?

**Comment by nepython** [comment]
@TheOneAboveAllTitan , I think `runflake8` and `runisort` files can be safely removed, no longer required. :smile:

File: .travis.yml (hunk 4 above)
Comment (TheOneAboveAllTitan) [reply .travis.yml:36]: I took reference from `openwisp-firmware-upgrader` and it is still present [there](https://github.com/openwisp/openwisp-firmware-upgrader/blob/6f798d429dffbf0d4a3503b4b3d1cb293706b2e7/.travis.yml#L34). Should I create a PR for that?

**Pull request merged**